LoopKit/CarbKit/DeletedCarbEntry.swift

```diff
@@ -9,21 +9,29 @@ import Foundation
 
 
 public struct DeletedCarbEntry {
-    public let externalID: String
+    public let externalID: String?
```

Comment:
Shouldn't this go away in favor of syncIdentifier?

Reply:
Yes, but there is an intermediate time for DIY Loop when they were running an older version and now they are running a version that contains this code were we support both `externalID` and `syncIdentifier`. It is just supporting the post-upgrade time period of a few days while all of the "old" `externalID` data is still cached locally.

Reply:
Hmm, not sure. I vaguely recall that the Nightscout endpoint for deleting via external ID versus syncIdentifer may be different. I'll look into it.

Reply:
While the `syncIdentifier` is a new field on the `DeletedCarbObject` model, it is not on the `CachedCarbObject` model. To keep symmetry between the two models, I'd prefer to keep this as-is.  Eventually we can just delete all reference to `externalID` in both since it won't be used at all.

We cannot migrate `externalID` to `syncIdentifier` in `CachedCarbObject` because both fields are used for different purposes in existing installations of Loop. Specifically, the `externalID` field is used as the external Nightscout identifier (once the carb object is uploaded), while the `syncIdentifier` field is used to accurately update or delete any given carb object in HealthKit.